### DISCUSSION TITLE
SDRPP as Panadapter? [RigCtl/Flrig]

### DISCUSSION THREAD

**Discussion by har5ha** [AlexandreRouma/SDRPlusPlus#753] [Q&A] [answered]
Hello All, 

I have a question regarding SDR++.  
Can it be used as a  Panadapter with RigCtl/FlRig (tuning the Yaesu Rig) to the frequency set on the SDR  and sync back and forth.   

TIA

**Comment by the-gobleo-tinkerer** [comment]
There is A RigCTL server module, which you need to enable in the Module Manager, that allows another program to change SDR++'s tuning. I'm not sure if it can sync back to the radio, but you can give it a try.

**Reply by har5ha** [reply]
Thanks! for your reply .  I noticed SDR++ has a "RigCTL server module" that allows another program(like GPredict) to change SDR++'s(HackRF) tuning. I cannot start the "RigCTL server module" inside SDR++ on the  port( '4532') as a the RigCtld (port '4532') is grabbed by RigCtld, during boot-up. But I can Telnet into "RigCTL server module" if its run on a different port(other than port '4532'). How do I make RigCtld and SDR++("RigCTL server module" ) talk to each other?. I am unable to figure that out. 

CubicSDR has this functionality of two-way sync between the Rig and the SDR(HackRF). I am wondering if this has been achieved in SDR++

**Reply by the-gobleo-tinkerer** [reply]
I'm not sure if you can, because the server module in SDR++ is essentially also a "RigCTLd": they accept tuning commands from a client and the client can also query the current frequency, but the server can't actively control the client. Someone requested a RigCTL client module in #174 but it's not implemented yet.

**Reply by har5ha** [reply]
Thanks again for clarifying. Can wait, for RigCTL Client to be implemented in SDR++. That would be amazing.

And a big thank you!, to all the developers involved in this wonderful project.

**Comment by n2iw** [comment, marked as answer]
You might want to look at my little tool [CAT-Relay](https://github.com/n2iw/cat-relay). It syncs frequency and mode between DXLab commander compatible logging software and SDR++. If you want to use flrig, there is a [fork](https://github.com/cdhowe/cat-relay) supports that. It has some small issues, that's why I haven't merge it in yet. You can use that version for the moment, until it's merged.

**Reply by har5ha** [reply]
Thanks! for the tips n2iw . Will definitely check them out. Appreciate it!. 73

**Reply by har5ha** [reply]
Thanks .. Once again n2iw! .. you saved the day!. I got "cat-relay" fork working perfectly.. I struggled with SDR++ syncing with my Rig, but then when I changed it to a mode that SDR++ has like FLRig (AM on both) it started syncing.. SDR++ didn't understand "Narrow-AM" which was being sent over by FLRig to SDR++ initially. I have to sit down and Map what Modes SDR++ understands, for which there is a section on Github page.. Also had to start SDR++ (demodulating signals) and then run the cat-relay..  73

**Reply by billybobnn5** [reply]
N2IW; Many thanks for the heads up on the flrig fork. I have tried to make the gqrxHamlib deriatives work off and on but too many refused connections & crashes. I run RumLogNG, Flrig, Fldigi and WSJT-X on macos so this tool is a natural for me. I ran the forked version tonight for much longer time than mine had ever run so will follow this activity. This is a great development to be able to have a sdr++ panadaptor. Sdr++ system loads is better than the other mac alternatives. It appears I can turn off sdr++ remote listening to let rig/flrig tune for xmit spot around pileup DX or work with second vfo in some fashion. The visual side of things really helps to compensate for some of my hearing loss.

**Reply by dkaye** [reply]
Using Python 2.7.16 I get the following error when running listen.py:
`  File "listen.py", line 62
    print(f'Connecting to Commander at {params[LOGGER_IP]}:{params[LOGGER_PORT]}')
                                                                                ^
SyntaxError: invalid syntax`

**Reply by n2iw** [reply]
> Using Python 2.7.16 I get the following error when running listen.py:
> ` File "listen.py", line 62 print(f'Connecting to Commander at {params[LOGGER_IP]}:{params[LOGGER_PORT]}') ^ SyntaxError: invalid syntax`

Python 2.x has been deprecated for a few years. Please use Python 3.x.
